### ISSUE TITLE
[BUG] Documentation of gap_threshold in generate_staypoints is incorrect

### ISSUE THREAD

**Issue by void-rooster** [mie-lab/trackintel#635]
The documentation:

> gap_threshold ([float](https://docs.python.org/3/library/functions.html#float), default 15.0 (minutes)) – The time threshold of determine whether a gap exists between consecutive pfs. Consecutive pfs with temporal gaps larger than ‘gap_threshold’ will be excluded from staypoints generation. Only valid in ‘sliding’ method.

I expect the test case in the code sample below to have one staypoint when the time threshold is 3 minutes, and ??? staypoints when the time threshold is 4 minutes.

I read:

> Consecutive pfs with temporal gaps larger than ‘gap_threshold’ will be excluded from staypoints generation.

as the positionfixes at `2025-01-01 00:04:30` and `2025-01-01 00:30:00` are excluded from staypoint generation, so I am inclined to think that according to the docs a 4 minute time threshold should result in 0 staypoints.

But a 3 minute time threshold results in 0 staypoints as well:

```python
from trackintel.preprocessing.positionfixes import generate_staypoints
import pandas as pd
import geopandas as gpd
from shapely.geometry import Point
   
user_id = 123
   
timestamps = [
   '2025-01-01 00:00:00', '2025-01-01 00:00:10', '2025-01-01 00:00:20', '2025-01-01 00:00:30',
   '2025-01-01 00:01:30', '2025-01-01 00:02:30', '2025-01-01 00:03:30', '2025-01-01 00:04:30',
   '2025-01-01 00:30:00', '2025-01-01 00:30:10', '2025-01-01 00:30:20', '2025-01-01 00:30:30'
]

location = [Point(0, 0)] * 8 + [Point(1, 1)] * 4

tracked_at = pd.to_datetime(timestamps).tz_localize("+00:00")

df = pd.DataFrame({
    'user_id': [user_id] * len(timestamps),
    'tracked_at': tracked_at,
    'geometry': location
})

gdf = gpd.GeoDataFrame(df, geometry='geometry', crs='EPSG:4326')

_, sps = generate_staypoints(gdf, dist_threshold=100, gap_threshold=15, time_threshold=4, include_last=True)
_, sps = generate_staypoints(gdf, dist_threshold=100, gap_threshold=15, time_threshold=3, include_last=True)
```

However this finds the staypoint:

```python
_, sps = generate_staypoints(gdf, dist_threshold=100, gap_threshold=24 * 60, time_threshold=5, include_last=True)
```

The documentation diverges from the implementation, [a reset of the window](https://github.com/mie-lab/trackintel/blob/master/trackintel/preprocessing/positionfixes.py#L491) being the likely culprit.

**Comment by hongyeehh** [comment]
Hi! 

A sp will only be created when the user leaves the area, defined by the time and distance threshold. In your example, this occurs between the two consecutive pfs at '2025-01-01 00:04:30' and '2025-01-01 00:30:00', and a new sp will be created with all pfs recorded before. But since these two pfs have a temporal gap of 25.5 minutes, setting `gap_threshold` smaller than this value will let the function ignore the creation of this sp. The reason is we have no info of where the user has been during this period, and assigning a sp with end time '2025-01-01 00:30:00' assumes the user is always stationary, which we would like to avoid.

Setting a larger `gap_threshold` implies one is more tolerant of data recording frequency. I hope this helps answer your concern.

**Comment by void-rooster** [comment]
I forgot that the positionfixes have to exit the staypoint to trigger its creation. Is there a chance of getting something like the `include_last` flag, but for staypoints immediately before a gap? This behavior is quite a blocker for us using `gap_threshold` properly, as we miss many, many staypoints when the staypoint end time has to be the time of the first positionfix outside the staypoint.

**Comment by hongyeehh** [comment]
Could you elaborate on the real-world use case for the proposed approach? 

I think it makes sense to set the sp end time as it is now because, as mentioned in Li et al. (2008), GPS signals tend to miss indoors. I am not sure about adding another parameter to the current already complicated implementation. But of course, we are open to support new staypoint detection methods with other assumptions.

**Comment by void-rooster** [comment]
I won't elaborate much on the data we create staypoints from---I will say that the positionfixes are noisy, with significant gaps of up to a few days (rarely). When we set `gap_threshold` to the timedelta of the largest gap, we get many more (reasonable) staypoints than if we set `gap_threshold` to 15 minutes, precisely because of the problem I've flagged---that the gaps immediately follow positionfixes that would qualify as a staypoint except for the gap + definition that a staypoint ends at the first positionfix outside the staypoint.

**Comment by void-rooster** [comment]
Right. The other pressure we have is that our data are Huge and algorithms like dbscan and st-dbscan may take too long; we will be looking into infostop soon. We like the Li et al. method of detecting staypoints, and the implementation in trackintel works well with our multiprocessing framework. What doesn't work for us are the adaptations done to the Li et al. algorithm to handle gaps.

**Comment by hongyeehh** [comment]
If you add new methods, feel free to open a new issue/PR. I am closing this because the original issue description is imprecise.
